### DISCUSSION TITLE
Filers in a cluster not in sync?

### DISCUSSION THREAD

**Discussion by michagruner** [seaweedfs/seaweedfs#3555] [Q&A]
Hello,

I have a cluster of seaweedfs with 3 filers. After I changed two of them from leveldb3 to rocksdb as a store (to be able to sync the cluster to another one) the filers are no longer in sync. Is there something I can do about this?

```
weedbackup@weedbackup:~$ cd weed2/
weedbackup@weedbackup:~/weed2$ ls
qubesbackup
weedbackup@weedbackup:~/weed2$ cd ..
weedbackup@weedbackup:~$ umount weed2/
weedbackup@weedbackup:~$ cd weed3/
weedbackup@weedbackup:~/weed3$ ls
data  home  homes  pve_spicetokenFamilyPC.sh  qubesbackup
weedbackup@weedbackup:~/weed3$ cd ..
weedbackup@weedbackup:~$ umount weed3/
weedbackup@weedbackup:~$ cd weed1
weedbackup@weedbackup:~/weed1$ ls
data  home  homes  nextcloud  pve_spicetokenFamilyPC.sh  qubesbackup

```

Thanks for your help,
Michael.

**Comment by chrislusf** [comment]
The filer store type needs to be the same.

**Reply by michagruner** [reply]
Adjusted the documentation accordingly: https://github.com/seaweedfs/seaweedfs/wiki/Filer-Store-Replication

**Comment by michagruner** [comment]
Thanks Chris for your answer. Switched the other filers off and tried to migrate the leveldb3 filer as described in https://github.com/seaweedfs/seaweedfs/wiki/Filer-Stores 

```
weedbackup@weedbackup:~$ ./weed shell -filer=weed1:8888
I0830 20:10:47.053748 masterclient.go:197 master localhost:9333 redirected to leader 192.168.1.122:9333
> cd /
> ls
data
home
homes
nextcloud
pve_spicetokenFamilyPC.sh
qubesbackup
topics
> meta.save
total 60121 directories, 440715 files
meta data for http://weed1:8888/ is saved to weed1:8888-20220830-201100.meta
> exit
weedbackup@weedbackup:~$ ./weed shell -filer=weed1:8888
I0830 20:13:33.442755 masterclient.go:197 master localhost:9333 redirected to leader 192.168.1.122:9333
> meta.load weed1:8888-20220830-201100.meta
error: CreateEntry : update entry /data: existing /data is a file
> 

```

Unfortunately I get this error after switching this one filer to rocksdb.

Do you have any idea what I may do wrong?

Thanks, Michael.

**Reply by chrislusf** [reply]
the source filer has `/data` as a directory.
the destination filer has `/data` as a file.

**Reply by michagruner** [reply]
Hm, I switched to rocksdb without an entry:

```
weedbackup@weedbackup:~$ ./weed shell -filer=weed1:8888
I0831 10:40:20.178479 masterclient.go:197 master localhost:9333 redirected to leader 192.168.1.122:9333
> ls
> meta.load weed1:8888-20220830-201100.meta
error: CreateEntry : update entry /data: existing /data is a file
> 
```
Is there something else to delete/alter when switching the filer-store?

Thanks, Michael

**Reply by chrislusf** [reply]
Could you please share the metadata file?

**Reply by michagruner** [reply]
sure, can I send it by mail somehow? I would mind puting it in the public space.

**Reply by chrislusf** [reply]
Thanks for the metadata file! It is a bug in filer RocksDB store. I added a fix a112cbe44c95d4685aefa43c27279e5aafb4b04c

**Reply by michagruner** [reply]
Thanks Chris, will give it a shot as soon as the next rocksDB docker is available.

**Reply by michagruner** [reply]
So the replay of the meta file with the rocksdb 3.25 works. Also all my filers (now all with rocksdb) are in sync again.

However, when I now try to sync to another cluster (rocksdb filer with volume data on a sshfs) it's awfully slow, it doesn't even sync any file at all. Is rocksdb (instead of leveldb) supposed to work with cluster-sync?

**Reply by chrislusf** [reply]
For `filer.sync`, there are no difference between leveldb and rocksdb.

**Comment by chrislusf** [comment]
the wiki change is wrong. I reverted it.

**Reply by michagruner** [reply]
Oh, sorry. My understanding from your first reply was the inconsistencies between the filers in my cluster were a result of mixing leveldb with rocksdb.
